src/poetry/console/commands/add.py

```diff
@@ -285,3 +299,22 @@ def notify_about_existing_packages(self, existing_packages: list[str]) -> None:
         for name in existing_packages:
             self.line(f"  • <c1>{name}</c1>")
         self.line(self._hint_update_packages)
+
+    def migrate_dev_dependencies(self, poetry_content: dict[str, Any]) -> None:
+        from poetry.factory import Factory
+
+        dev_group = poetry_content["group"]["dev"]["dependencies"]
```

Comment:
This seems more elaborate than is needed. Is there a reason that we can't do a more direct assignment?

Reply:
I noticed that with older versions of poetry the dependencies in `pyproject.toml` were not always in canonical form -- e.g, `Sphinx` and `PyGitHub`. `poetry add` doesn't seem to recognize these.

Reply:
I'm still wondering if there should be a command to explicitly do the migration.

Reply:
That command would be able to deal with things like making sure the dependencies are in canonical form.

Reply:
I think this may be necessary since the group `dev` doesn't seem to exist when `dev-dependencies` are used:
```python
(Pdb) 
> /home/rzuckerm/git/public/poetry/src/poetry/console/commands/add.py(234)handle()
-> self.poetry.package.dependency_group(group).remove_dependency(
(Pdb) self.poetry.package.dependency_group(group)
*** ValueError: The dependency group "dev" does not exist.
```

Reply:
I'm really wondering what's in `poetry_content` -- can we perform a direct reassignment instead of trying to re-create it from scratch?

Reply:
For this test:
```
test_add_existing_constraint_old_dev_section_migrates[cachy]
```
This is what is in `poetry_content`:
```python
$ poetry run pytest -vvl -k old_dev_section
Test session starts (platform: linux, Python 3.10.6, pytest 7.1.3, pytest-sugar 0.9.5)
cachedir: .pytest_cache
Using --randomly-seed=3658096042
rootdir: /home/rzuckerm/git/public/poetry
plugins: xdist-2.5.0, forked-1.4.0, cov-3.0.0, randomly-3.12.0, mock-3.8.2, sugar-0.9.5
collecting ... 
>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>
> /home/rzuckerm/git/public/poetry/src/poetry/console/commands/add.py(301)migrate_dev_dependencies()
-> dev_group = poetry_content["group"]["dev"]["dependencies"]
(Pdb) poetry_content
{'name': 'simple-project', 'version': '1.2.3', 'description': 'Some description.', 'authors': ['Sébastien Eustace <sebastien@eustace.io>'], 'license': 'MIT', 'readme': ['README.rst'], 'homepage': 'https://python-poetry.org', 'repository': 'https://github.com/python-poetry/poetry', 'documentation': 'https://python-poetry.org/docs', 'keywords': ['packaging', 'dependency', 'poetry'], 'classifiers': ['Topic :: Software Development :: Build Tools', 'Topic :: Software Development :: Libraries :: Python Modules'], 'dependencies': {'python': '~2.7 || ^3.4'}, 'scripts': {'foo': 'foo:bar', 'baz': 'bar:baz.boom.bim', 'fox': 'fuz.foo:bar.baz'}, 'dev-dependencies': {'cachy': '^0.2.0'}, 'group': {'dev': {'dependencies': {}}}}
```

Also, as I showed above, the `dev` group does not yet exist. If I just do a direct assignment, then my new tests fail.